packages/@aws-cdk/aws-codepipeline-actions/lib/s3/deploy-action.ts

```diff
@@ -86,6 +87,14 @@ export interface S3DeployActionProps extends codepipeline.CommonAwsActionProps {
    * @default - none, decided by the HTTP client
    */
   readonly cacheControl?: CacheControl[];
+
+  /**
+   * The AWS KMS encryption key for the host bucket.
+   * The encryptionKey parameter encrypts uploaded artifacts with the provided AWS KMS key.
+   * For a KMS key, you can use the key ID, the key ARN, or the alias ARN.
```

Comment:
When providing an `IKey` the user wouldn't use a key ID, a key ARN, or an alias. This line should be removed.